### DISCUSSION TITLE
brew bump --open-pr bettertouchtool fails

### DISCUSSION THREAD

**Discussion by xyzzy-plugh-plover** [Homebrew/discussions#3814] [Everyday usage] [answered]
`brew bump --open-pr bettertouchtool` fails on macOS 12.6 with the following output:

```
brew bump --open-pr bettertouchtool
==> bettertouchtool
Current cask version   :  3.913,2075
Latest livecheck version: 3.914,2076
Latest Repology version:  present only in Homebrew
Open pull requests:       none
/usr/local/Homebrew/Library/Homebrew/shims/ruby/bundle:15:in `load': cannot load such file -- /usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/bundler-2.3.24-x86_64-darwin-21/exe/bundle (LoadError)
	from /usr/local/Homebrew/Library/Homebrew/shims/ruby/bundle:15:in `<main>'
Error: failed to run `/usr/local/Homebrew/Library/Homebrew/shims/ruby/bundle install`!
```

A directory for bundler exists named `bundler-2.3.24` instead of being named `bundler-2.3.24-x86_64-darwin-21`. i.e. the following directory exists:

```
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/bundler-2.3.24/exe/bundle 
```

Info:

```
brew doctor
Your system is ready to brew.

brew config
HOMEBREW_VERSION: 3.6.6-32-g2bec760
ORIGIN: https://github.com/Homebrew/brew.git
HEAD: 2bec760529343832be00662ed99151ddbf41b486
Last commit: 3 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 590d65b8f417e1373fe2240633da4b606c34832d
Core tap last commit: 67 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.3L7XAFxtwf/org.xquartz:0
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: quad-core 64-bit haswell
Clang: 14.0.0 build 1400
Git: 2.38.1 => /usr/local/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.6-x86_64
CLT: 14.0.0.0.1.1661618636
Xcode: 14.0.1
```

**Comment by SMillerDev** [comment]
@Bo98 any ideas?

**Reply by xyzzy-plugh-plover** [reply]
I'm happy to provide whatever else is necessary to investigate this.

**Reply by Bo98** [reply]
Sorry missed this ping for some reason. Does deleting `/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/specifications` fix the issue?

If you can take a backup of that folder first, and if deleting it works, then send me the contents of it so I can have a look at what might have happened.

If deleting it doesn't work, let me know and I'll see what else might be the issue.

**Reply by Bo98** [reply]
Can you also show the output of `brew ruby -e 'puts Gem::VERSION'`?

**Reply by xyzzy-plugh-plover** [reply]
```
$ brew ruby -e 'puts Gem::VERSION' 
3.0.3.1
```

Ran that without deleting the `specifications` directory. Will try that now.

**Reply by xyzzy-plugh-plover** [reply]
Still failed after deleting the `specifications` directory:

```
$ brew bump --open-pr bettertouchtool                                
==> bettertouchtool
Current cask version   :  3.924,2087
Latest livecheck version: 3.927,2090
Latest Repology version:  present only in Homebrew
Open pull requests:       none
==> Installing 'bundler' gem
/usr/local/Homebrew/Library/Homebrew/shims/ruby/bundle:15:in `load': cannot load such file -- /usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/bundler-2.3.24-x86_64-darwin-21/exe/bundle (LoadError)
	from /usr/local/Homebrew/Library/Homebrew/shims/ruby/bundle:15:in `<main>'
Error: failed to run `/usr/local/Homebrew/Library/Homebrew/shims/ruby/bundle install`!
```

**Reply by xyzzy-plugh-plover** [reply]
```
$ diff bundler-2.3.24.old.gemspec bundler-2.3.24.gemspec 
21c21
<   s.rubygems_version = "3.2.5".freeze
---
>   s.rubygems_version = "3.0.3.1".freeze
24c24
<   s.installed_by_version = "3.2.5" if s.respond_to? :installed_by_version
---
>   s.installed_by_version = "3.0.3.1" if s.respond_to? :installed_by_version
```

FYI: I only use the macOS built-in Ruby:

```
$ whence -p ruby
/usr/bin/ruby

$ ruby --version
ruby 2.6.10p210 (2022-04-12 revision 67958) [universal.x86_64-darwin21]

$ whence -p gem
/usr/bin/gem

$ gem --version
3.2.5
```

Homebrew obviously installs some other version of gem. Maybe Homebrew installs some other Ruby. I don't know of any other Ruby installs.

**Reply by Bo98** [reply]
Ok. Default RubyGems on macOS is 3.0.3.1, so my guess the 3.2 might have caused issues. I recently shipped a change which forced system RubyGems to be used, which appears to be working given how the spec has changed. Though apparently not enough to fix the problem you were having.

I've not managed to reproduce the problem yet, but I do have another fix that I'm working on with upstream. I'll get back to you in a few days once that's progressed.

**Comment by pthariensflame** [comment]
I've run into this same error now, although it wants `/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/bundler-2.3.24-x86_64-darwin-22/exe/bundle`.  It interrupts anything involving `brew style` or `brew audit`.

**Reply by xyzzy-plugh-plover** [reply]
Are you on macOS 13? I'm now on 12.6.1, which still mentions `darwin-21`, just like what was mentioned when I was on macOS 12.6 when I opened this discussion.

**Comment by Bo98** [comment, marked as answer]
I've pushed some fixes that will hopefully cover this. Let me know if you are still having issues after a `brew update`.

**Reply by xyzzy-plugh-plover** [reply]
Thanks. Will try once I have a formula or cask that fails its livecheck.
